cmd/server/server.go

```diff
@@ -191,14 +192,18 @@ func (s *server) startService() common.Daemon {
 			params.BlobstoreClient, err = filestore.NewClient(s.cfg.Archival.Filestore)
 		}
 		if s.cfg.Archival.S3store != nil {
-			s3cli, err := s3store.ClientFromConfig(s.cfg.Archival.S3store)
-			if err != nil {
+			var s3cli s3iface.S3API
+			s3cli, err = s3store.ClientFromConfig(s.cfg.Archival.S3store)
+			if err == nil {
 				params.BlobstoreClient = s3store.NewClient(s3cli)
 			}
 		}
 		if err != nil {
 			log.Fatalf("error creating blobstore: %v", err)
 		}
+		if params.BlobstoreClient == nil {
```

Comment:
I think we should perform this check (if neither fliestore nor s3 store is specified) on L190.

Reply:
+1 to that.

Reply:
Changde in 12c430e0172fd8446861626da6aec7f32b693f17